PIVX-Labs/MyPIVXWallet
# Shield Activity

## Abstract

Implements shield activity, i.e. shield transactions regarding the wallet are decrypted and loaded in the dashboard activity.

 3 new transactions types have been added:

- Shielding, which is a self (t -> s) transaction;
- De-Shielding, which is a self (s->t) transaction;
- Shield sent to self, which is a self (s->s) transaction.

![shield_activity](https://github.com/user-attachments/assets/72d0ba64-df10-44f3-9baa-00c7fa8efbd8)


  Transactions received and sent to external wallets are marked as before, with the addition of shield addresses:
- Sent to (transparent or shield address);
- Received with (transparent or shield address)


I suggest to review the PR commit by commit, is easier than looking at the overall diff. Also to test the PR you have to manually build and use the following version of the pivx-shield library: [PR 81](https://github.com/PIVX-Labs/pivx-shield/pull/81).


---

## Testing
To test this PR, it's suggested to attempt these user flows, or variations of these:
- Create and sync a shield wallet;
- Create the 3 new types of transaction;
- Send/Receive shielded pivs from an external wallet;
- Close and re-open the wallet: the shield activity should be unchanged;
- Resync the wallet from zero: the shield activity should be unchanged.
